### PR TITLE
feat: Add TooltipIcon Component

### PR DESCRIPTION
**Pull request by marguslant0**
**Changes:**
- Extracting the logic of tooltip and the icon that triggers it out of `InputLabel` and into a separate `TooltipIcon` component
- Using the newly created `TooltipIcon` component inside `InputLabel` in place of the previous code
- Allowing the new component to set the size of the icon
- Updating tests

Storybook: http://localhost:6006/?path=/story/components-tooltipicon--with-default-icon-size

### CODE REVIEW

File: src/components/TooltipIcon/TooltipIcon.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    it("should render an icon with custom size", () => {
+        mount(<TooltipIcon tooltip={TOOLTIP_PROPS} iconSize={IconSize.Size20} />);
+
+        cy.get(TOOLTIP_ICON_ID).find(TOOLTIP_ICON_TRIGGER_ID).children().should("have.class", "tw-h-5");
+    });
+
+    it("should render a tooltip when the icon is hovered", () => {
+        mount(<TooltipIcon tooltip={TOOLTIP_PROPS} />);
+
+        cy.get(TOOLTIP_ICON_ID).realHover({ position: "top" });
```
Comment (jonasbuechel) [inline src/components/TooltipIcon/TooltipIcon.spec.tsx:31]:
```suggestion
        cy.get(TOOLTIP_ICON_TRIGGER_ID).realHover({ position: "top" });
```

**Review by jonasbuechel — CHANGES_REQUESTED** [review]
The change looks fine, just the test did not run so I added a suggestion 👌

File: src/components/TooltipIcon/TooltipIcon.spec.tsx (hunk 1 above)
Comment (marguslant0) [reply src/components/TooltipIcon/TooltipIcon.spec.tsx:31]: Thanks for the review, @jonasbuechel! 💪 

Using `TOOLTIP_ICON_TRIGGER_ID`, can you get the test to pass every time? I tried but got flaky results, and I'm trying now to understand what causes it 🤷

Comment (marguslant0) [reply src/components/TooltipIcon/TooltipIcon.spec.tsx]: Looks like focus can randomly get lost in some cases. To ensure it won't, I added a click on the entire "body" before the hover action is performed. This seems to make it stable.

Comment (jonasbuechel) [reply src/components/TooltipIcon/TooltipIcon.spec.tsx]: I think it also depends if you run the tests headless (`npm run test`) or with an open browser (`npm run test:browser`) 🤔

**Review by jonasbuechel — APPROVED** [review]
I was able to run the tests 3x with open and with closed browser and it's passing now 👍

**Pull request merged**